### ISSUE TITLE
Concise description of the issue

### ISSUE THREAD

**Issue by imjasonh** [sigstore/sigstore#436]
Followups from https://github.com/sigstore/sigstore/pull/435

- [ ] remove `panic` in `fulcioroots.Get()`
- [ ] add doc comments for un-doc-commented exported methods (or unexport them if appropriate)

/assign @haydentherapper

**Comment by znewman01** [comment]
Quoting from 
From https://github.com/sigstore/cosign/issues/1865:

>> I don't think I'd mind as a prospective consumer of the package that its behavior was dependent on an env var, especially if that was named like SIGSTORE_ and not COSIGN_.
>
> I would TBH, I think overall burying config options deep in some automagic code that's tough to modify as a caller without setting an env var is bad for testability etc. Further, if I'm using sigstore as a client but I want my own TUF state it'd possible that I wouldn't want to respect that variable. Maybe if there was a Sigstore-TUF client struct that had this configurable, and it had a default constructor that pulled from an environment variable?
>
> And while I'm being cranky, I already think this is misbehaved because it doesn't respect XDG_STATE_HOME too.

So I'm gonna add to your TODO list :)

**Comment by znewman01** [comment]
Also fine to stick one that *does* consult the env var in sigstore/sigstore as long as it's not the only option. I just don't want to force that on callers.

**Comment by imjasonh** [comment]
> Also fine to stick one that _does_ consult the env var in sigstore/sigstore as long as it's not the only option. I just don't want to force that on callers.

If more callers want to adopt the env var version, we can move that into sigstore/sigstore too, as a separate opt-in process. For now I think you've convinced me that the env var behavior is only documented for cosign, and so should stay there unless/until someone else wants it.

After some of this lands I want to move more things into `internal` packages in cosign, to discourage/prevent their use outside of cosign and encourage folks to depend on sigstore/sigstore

**Comment by haydentherapper** [comment]
Are we concerned with breaking existing clients that may rely on this now? Hopefully folks are using TUF and not relying on the environment variables, but those were added for testing purposes. There's one for each of the TUF targets (CT log, Rekor, Fulcio), so we should probably add support for passing in the env var to a constructor (or maybe adding it to the context?) in a later PR.

**Comment by znewman01** [comment]
Hmm...I was gonna say "it's not documented anywhere" but [that's not true](https://docs.sigstore.dev/cosign/openid_signing/). I dunno, I guess I struggle to think of a situation that this would break...cosign will continue to work, and callers of cosign will probably continue to work too if we handle the env var in cosign itself. 🤷

**Comment by haydentherapper** [comment]
Sorry, I didn’t mention context - the PR in Sigstore/Sigstore removed the environment variable. I just wanted to confirm we plan to continue supporting the variable by adding a constructor or something.

**Comment by imjasonh** [comment]
I agree with Zack, callers of cosign methods will continue to get the same behavior with the env var, for better or worse. If they move to the s/s equivalent (as we recommend) they won't, but I doubt they've intended to get that behavior anyway. If they need it, they can stay on the cosign version until the s/s one gets an equivalent.

**Comment by haydentherapper** [comment]
Sounds good, let’s add this as a TODO on this issue to track this.

**Comment by haydentherapper** [comment]
Bumping the root doesn't need to happen right now, it just saves one or two network calls to fetch an extra root.
